drivers/led/issi/is31fl3729-mono.h

```diff
@@ -0,0 +1,271 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "progmem.h"
+#include "util.h"
+
+#define IS31FL3729_REG_SCALING 0x90
+#define IS31FL3729_REG_CONFIGURATION 0xA0
+#define IS31FL3729_REG_GLOBALCURRENT 0xA1
+#define IS31FL3729_REG_PULLDOWNUP 0xB0
+#define IS31FL3729_REG_SPREAD_SPECTRUM 0xB1
+#define IS31FL3729_REG_PWM_FREQUENCY 0xB2
+#define IS31FL3729_REG_RESET 0xCF
+
+#define IS31FL3729_I2C_ADDRESS_GND 0x34
+#define IS31FL3729_I2C_ADDRESS_SCL 0x35
+#define IS31FL3729_I2C_ADDRESS_SDA 0x36
+#define IS31FL3729_I2C_ADDRESS_VCC 0x37
+
+#if defined(LED_MATRIX_IS31FL3729)
+#    define IS31FL3729_LED_COUNT LED_MATRIX_LED_COUNT
+#endif
+
+#if defined(IS31FL3729_I2C_ADDRESS_4)
+#    define IS31FL3729_DRIVER_COUNT 4
+#elif defined(IS31FL3729_I2C_ADDRESS_3)
+#    define IS31FL3729_DRIVER_COUNT 3
+#elif defined(IS31FL3729_I2C_ADDRESS_2)
+#    define IS31FL3729_DRIVER_COUNT 2
+#elif defined(IS31FL3729_I2C_ADDRESS_1)
+#    define IS31FL3729_DRIVER_COUNT 1
+#endif
+
+typedef struct is31fl3729_led_t {
+    uint8_t driver : 2;
+    uint8_t v;
+} PACKED is31fl3729_led_t;
+
+extern const is31fl3729_led_t PROGMEM g_is31fl3729_leds[IS31FL3729_LED_COUNT];
+
+void is31fl3729_init_drivers(void);
+void is31fl3729_init(uint8_t addr);
+void is31fl3729_write_register(uint8_t addr, uint8_t reg, uint8_t data);
+bool is31fl3729_write_pwm_buffer(uint8_t addr, uint8_t *pwm_buffer);
+
+void is31fl3729_set_value(int index, uint8_t value);
+void is31fl3729_set_value_all(uint8_t value);
+
+void is31fl3729_set_scaling_register(uint8_t index, uint8_t value);
+
+// This should not be called from an interrupt
+// (eg. from a timer interrupt).
+// Call this while idle (in between matrix scans).
+// If the buffer is dirty, it will update the driver with the buffer.
+void is31fl3729_update_pwm_buffers(uint8_t addr, uint8_t index);
+void is31fl3729_update_scaling_registers(uint8_t addr, uint8_t index);
+
+void is31fl3729_flush(void);
+
+// Noise reduction using Spread Spectrum register
+#define IS31FL3729_SSP_DISABLE 0x00
+#define IS31FL3729_SSP_5_1980US 0x10
+#define IS31FL3729_SSP_5_1200US 0x11
+#define IS31FL3729_SSP_5_820US 0x12
+#define IS31FL3729_SSP_5_660US 0x13
+#define IS31FL3729_SSP_15_1980US 0x14
+#define IS31FL3729_SSP_15_1200US 0x15
+#define IS31FL3729_SSP_15_820US 0x16
+#define IS31FL3729_SSP_15_660US 0x17
+#define IS31FL3729_SSP_24_1980US 0x18
+#define IS31FL3729_SSP_24_1200US 0x19
+#define IS31FL3729_SSP_24_820US 0x1A
+#define IS31FL3729_SSP_24_660US 0x1B
+#define IS31FL3729_SSP_34_1980US 0x1C
+#define IS31FL3729_SSP_34_1200US 0x1D
+#define IS31FL3729_SSP_34_820US 0x1E
+#define IS31FL3729_SSP_34_660US 0x1F
```

Comment:
Please split this up into separate defines for each field, so they can be ORed together.